application/src/main/java/org/opentripplanner/apis/APIUtils.java

```diff
@@ -0,0 +1,62 @@
+package org.opentripplanner.apis;
+
+import io.micrometer.core.instrument.Tag;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MultivaluedMap;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+public final class APIUtils {
+
+  /**
+   * This method tries to find a tracing tag from a request's headers.
+   * If no value is found for a tag, the value is set to "__UNKNOWN__".
+   *
+   * @param tracingHeaderTags a collection of tag names to match against headers
+   * @param headers headers from a request
+   * @return a list of tracing tags with computed values
+   */
+  public static Iterable<Tag> getTagsFromHeaders(
```

Comment:
According to the naming conventions this should be `find*`, so probably `findTagsInHeaders`.